daemon/src/wallet.rs

```diff
@@ -0,0 +1,50 @@
+use crate::model::{WalletError, WalletInfo};
+use anyhow::Result;
+use bdk::bitcoin::Amount;
+use bdk::blockchain::{Blockchain, NoopProgress};
+use bdk::database::BatchDatabase;
+use bdk::wallet::AddressIndex;
+use tokio::sync::watch;
+
+pub fn wallet_sync<B, D>(
+    wallet_feed_sender: &watch::Sender<WalletInfo>,
+    wallet: &bdk::Wallet<B, D>,
+    previous: WalletInfo,
+) -> Result<Option<WalletInfo>, WalletError>
+where
+    B: Blockchain,
+    D: BatchDatabase,
+{
+    wallet
+        .sync(NoopProgress, None)
+        .map_err(|_| WalletError::Sync)?;
+
+    let balance = wallet.get_balance().map_err(|_| WalletError::Balance)?;
+
+    let update = match previous {
+        WalletInfo::Update {
+            balance: prev_balance,
+            ..
+        } => balance != prev_balance.as_sat(),
+        WalletInfo::Syncing | WalletInfo::Error(_) => true,
+    };
+
+    let wallet_info = if update {
+        let address = wallet
+            .get_address(AddressIndex::New)
```

Comment:
Why not `LastUnused`?

Reply:
We only trigger this code path if either the balance changed or we ran into an Error or were syncing before.
We might want to distinguish those cases, but I felt we should not run into the Error and Syncing path too often (initially and upon error...), so I thought New would be find. Did not give it much extra though.

I we use `LastUsed`, wouldn't this result in address reuse?

Reply:
It is Last::**Un**used!